quote-server/server.js

```diff
@@ -1,32 +1,59 @@
-// server.js
-// This is where your node app starts
-
-//load the 'express' module which makes writing webservers easy
 import express from "express";
-//load the quotes JSON
-import quotes from "./quotes.json" assert { type: "json" };
+import crypto from "crypto";
+import cors from "cors";
 
 const app = express();
-// Now register handlers for some routes:
-//   /                  - Return some helpful welcome info (text)
-//   /quotes            - Should return all quotes (json)
-//   /quotes/random     - Should return ONE quote (json)
-app.get("/", (request, response) => {
-  response.send("Neill's Quote Server!  Ask me for /quotes/random, or /quotes");
+
+let quotes = [];
+
+app.use(cors());
+
+app.use(async (req, res, next) => {
```

Comment:
What does `app.use` here do? What will cause the function you pass to it to get called? How often will it get called?

Reply:
It is used to execute middleware functions before each route method gets executed.

Reply:
Interesting! What are the trade-offs between doing this for every request in a middleware, vs doing it once when the server first starts up?

Reply:
Ahh I see now, yes. It would be more efficient if it gets the data once. There is no need to fetch it every time. Thank you very much.

Reply:
I have updated the code using a function once in the beginning.